fe/fe-core/src/main/java/com/starrocks/http/rest/TableQueryPlanAction.java

```diff
@@ -142,8 +144,16 @@ protected void executeWithoutPassword(BaseRequest request, BaseResponse response
                 throw new StarRocksHttpException(HttpResponseStatus.BAD_REQUEST,
                         "POST body must contains [sql] root object");
             }
-            LOG.info("receive SQL statement [{}] from external service [ user [{}]] for database [{}] table [{}]",
-                    sql, ConnectContext.get().getCurrentUserIdentity(), dbName, tableName);
+
+            String warehouseName = WarehouseManager.DEFAULT_WAREHOUSE_NAME;
+            if (request.getRequest().headers().contains(WAREHOUSE_KEY)) {
+                warehouseName = request.getRequest().headers().get(WAREHOUSE_KEY);
+                Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().getWarehouse(warehouseName);
+                ConnectContext.get().setCurrentWarehouseId(warehouse.getId());
```

Comment:
if the warehousename is provided, the warehouse id set in connectcontext, otherwise it is empty. Better keep the behavior consistent no matter the warehousename is provided or not from the http header.